runtime/timelib_wrapper.cpp

```diff
@@ -215,3 +225,511 @@ std::pair<int64_t, bool> php_timelib_strtotime(const string &tz_name, const stri
 
   return {ts, true};
 }
+
+static timelib_error_container *last_errors_global = nullptr;
+
+using ScriptMemGuard = decltype(std::function{make_malloc_replacement_with_script_allocator})::result_type;
+
+// NB: should be called under script allocator, because of calls to free() inside timelib_error_container_dtor()
+static void update_errors_warnings(timelib_error_container *last_errors, [[maybe_unused]] const ScriptMemGuard &guard) {
+  if (last_errors_global) {
+    timelib_error_container_dtor(last_errors_global);
+    last_errors_global = nullptr;
+  }
+  last_errors_global = last_errors;
+}
+
+void free_timelib() {
+  auto script_guard = make_malloc_replacement_with_script_allocator();
+  update_errors_warnings(nullptr, script_guard);
+}
+
+static string gen_parse_error_msg(const timelib_error_container &err, const string &str) {
+  string error_msg{"Failed to parse time string "};
+  error_msg.append(1, '(').append(str).append(1, ')');
+  error_msg.append(" at position ").append(err.error_messages[0].position);
+  error_msg.append(" (").append(1, err.error_messages[0].character).append("): ");
+  error_msg.append(err.error_messages[0].message);
+  return error_msg;
+}
+
+static const string NOW{"now"};
+
+std::pair<timelib_time *, string> php_timelib_date_initialize(const string &tz_name, const string &time_str, const char *format) {
+  auto script_guard = make_malloc_replacement_with_script_allocator();
+
+  const string &time_str_new = time_str.empty() ? NOW : time_str;
+  timelib_error_container *err = nullptr;
+  timelib_time *t = format
+    ? timelib_parse_from_format(format, time_str.c_str(), time_str.size(), &err, timelib_builtin_db(), timelib_parse_tzfile)
+    : timelib_strtotime(time_str_new.c_str(), time_str_new.size(), &err, timelib_builtin_db(), timelib_parse_tzfile);
+
+  /* update last errors and warnings */
```

Comment:
I'm not sure it's common to use `/**/` style  comments in our codebase (except for some top-level doc comments).

Reply:
I prefer `//` comments too, `/**/` appears just from copy-paste, fixed.